assets/JS/index.js

```diff
@@ -13,16 +13,27 @@ $(document).ready(function () {
         .then(function (data) {
             console.log(data);
             const categoryArray = data.trivia_categories;
-            $.each(categoryArray, function (i) {
+            categoryArray.forEach(function(category) {
                 categoryInput.append(
-                    $('<option></option>').val(categoryArray[i].id).html(categoryArray[i].name)
+                  $('<option></option>').val(category.id).html(category.name)
                 );
             });
             $('select').formSelect();
-        });
-        submitButton.addEventListener("click", submitModalForm);
-        
-});
+        })
+        .catch(function (error) {
+          console.error('Error fetching categories', error)
+        })
+        categoryInput.on('change', function () {
+          const selectedCategory = $(categoryInput).val();
+          localStorage.setItem('selectedCategory', selectedCategory);
+      });
+      $('.submit').on('click', submitModalForm);
```

Comment:
Pls change variable in line #1. 
Instead of current: `const submitButton = document.querySelector(".submit");`
do this `const submitButton = $('.submit')`
Then reuse it here, e.g.
`submitButton.on('click', submitModalForm);`